plugins/block-shareable-procedures/src/blocks.ts

```diff
@@ -417,25 +417,29 @@ const procedureDefMutator = {
 
   /**
    * Returns the state of this block as a JSON serializable object.
+   * @param doFullSerialization Tells the block if it should serialize
+   *     its entire state (including data stored in the backing procedure
+   *     model). Used for copy-paste.
    * @returns The state of this block, eg the parameters and statements.
    */
-  saveExtraState: function() {
+  saveExtraState: function(doFullSerialization) {
     const state = Object.create(null);
     state['procedureId'] = this.getProcedureModel().getId();
 
-    const params = this.getProcedureModel().getParameters();
-    if (!params.length && this.hasStatements_) return state;
-
-    if (params.length) {
-      state['params'] = params.map((p) => {
-        return {
-          'name': p.getName(),
-          'id': p.getVariableModel().getId(),
-          // Ideally this would be id, and the other would be varId,
-          // but backwards compatibility :/
-          'paramId': p.getId(),
-        };
-      });
+    if (doFullSerialization) {
+      state['fullSerialization'] = true;
+      const params = this.getProcedureModel().getParameters();
+      if (params.length) {
+        state['params'] = params.map((p) => {
+          return {
+            'name': p.getName(),
+            'id': p.getVariableModel().getId(),
+            // Ideally this would be id, and the other would be varId,
+            // but backwards compatibility :/
+            'paramId': p.getId(),
+          };
+        });
+      }
     }
     if (!this.hasStatements_) {
```

Comment:
Why do you only need to set this when `hasStatements_` is false, rather than always?

Reply:
Default is true. We always skip serialization of defaults (or we try to anyways) because it keeps the save file size smaller. E.g. we also don't serialize a `'params'` array if the procedure has no params.